website/src/components/SearchPage/fields/AutoCompleteField.tsx

```diff
@@ -59,106 +88,272 @@ export const AutoCompleteField = ({
         return allMatchedOptions.slice(0, maxDisplayedOptions);
     }, [options, query, maxDisplayedOptions]);
 
+    const handleChange = (value: string | string[] | null) => {
+        if (!multiSelect) {
+            // Single-select mode - convert NULL_QUERY_VALUE to null
+            const singleValue = Array.isArray(value) ? value[0] : value;
+            const finalValue = singleValue === NULL_QUERY_VALUE ? null : (singleValue ?? '');
+            setSomeFieldValues([field.name, finalValue]);
+            return;
+        }
+
+        // Multi-select mode - value is an array when multiple=true
+        if (Array.isArray(value)) {
+            if (value.length === 0) {
+                setSomeFieldValues([field.name, '']);
+            } else {
+                // Convert NULL_QUERY_VALUE to null for consistency
+                const convertedValues = value.map((v) => (v === NULL_QUERY_VALUE ? null : v));
+                setSomeFieldValues([field.name, convertedValues]);
+            }
+        }
+    };
+
+    const handleClear = () => {
+        setQuery('');
+        if (multiSelect) {
+            // For multi-select, pass an empty array to handleChange
+            handleChange([]);
+        } else {
+            // For single-select, clear normally
+            setSomeFieldValues([field.name, '']);
+        }
+    };
+
+    // Convert selectedValues Set to array for multi-select Combobox value
+    const multiSelectValue = useMemo(() => {
+        if (!multiSelect) return undefined;
```

Comment:
Are we ok with this syntax here? I thought usually we use brackets around the body of an `if`?
```suggestion
        if (!multiSelect) { 
          return undefined;
        }
```